core/common/src/main/java/alluxio/retry/RetryUtils.java

```diff
@@ -47,10 +50,28 @@ public static void retry(String action, RunnableThrowsIOException f, RetryPolicy
   }
 
   /**
-   * @return the best effort policy with no retry
+   * Notice we assume the exception with the method is retryable, so we are not wasting time here.
+   * Retries the given retryable method until it succeed or the retry policy expires. If the retry
+   * policy expires, the last exception generated will be rethrown as a AlluxioRuntimeException.
+   *
+   * @param description a description of the function
+   * @param f the function to retry which returns value
+   * @param policy the retry policy to use
+   * @param <V> result type returned by callable
+   * @return callable result
    */
-  public static RetryPolicy noRetryPolicy() {
-    return new CountingRetry(0);
+  public static <V> V retryCallable(String description, Callable<V> f, RetryPolicy policy) {
```

Comment:
You can just call this `retry`. I don't think you need both this one and the `retry` above. They are essentially the same.

Reply:
the semantic is kind of different since retry throw IOException and remove handling IOException involve some extra work. I would have another PR combining these two